src/test/java/org/dataloader/DataLoaderTest.java

```diff
@@ -78,9 +88,36 @@ public void should_Build_a_really_really_simple_data_loader() {
     }
 
     @Test
-    public void should_Support_loading_multiple_keys_in_one_call() {
+    public void basic_map_batch_loading() {
+        MappedBatchLoader<String, String> evensOnlyMappedBatchLoader = (keys) -> {
+            Map<String, String> mapOfResults = new HashMap<>();
+
+            AtomicInteger index = new AtomicInteger();
+            keys.forEach(k -> {
+                int i = index.getAndIncrement();
+                if (i % 2 == 0) {
+                    mapOfResults.put(k, k);
+                }
+            });
+            return CompletableFuture.completedFuture(mapOfResults);
+        };
+        DataLoader<String, String> loader = DataLoaderFactory.newMappedDataLoader(evensOnlyMappedBatchLoader);
+
+        loader.load("A");
+        loader.load("B");
+        loader.loadMany(asList("C", "D"));
+
+        List<String> results = loader.dispatchAndJoin();
+
+        assertThat(results.size(), equalTo(4));
+        assertThat(results, equalTo(asList("A", null, "C", null)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataLoaderFactories")
```

Comment:
We now run this test (and any test annotated with `@MethodSource("dataLoaderFactories")`) with all the `TestDataLoaderFactory` instances returned from the static method `dataLoaderFactories`.